packages/darklang/languageTools/parser/typeDeclaration.dark

```diff
@@ -0,0 +1,178 @@
+module Darklang =
+  module LanguageTools =
+    module Parser =
+      module TypeDeclaration =
+        let parseDefinition
+          (node: ParsedNode)
+          : Stdlib.Result.Result<WrittenTypes.TypeDeclaration.Definition, WrittenTypes.Unparseable> =
+          if node.typ == "type_decl_def" then
+            match node.children with
+            | [ child ] when child.typ == "type_decl_def_alias" ->
+              match child.children with
+              | [ single ] when single.typ == "type_reference" ->
+                let typRef = TypeReference.parse single
+
+                match typRef with
+                | Ok typRef ->
+                  (WrittenTypes.TypeDeclaration.Definition.Alias typRef)
+                  |> Stdlib.Result.Result.Ok
+
+                | Error _ -> createUnparseableError single
+              | _ -> createUnparseableError child
+
+            | [ child ] when child.typ == "type_decl_def_record" ->
+              let openBraceNode = findField child "symbol_open_brace"
```

Comment:
I think the details of this branch (and similarly, for enums) could be extracted to a nice helper fn above